### PR TITLE
Adds wooden chapel pews

### PR DESCRIPTION
**Pull request by Caiggas**
pewscreenshotfinal

## About The Pull Request

This adds wooden chapel pews to the game. They function much like ratty couches, in that there are several different pieces which can be mixed and matched to make different length pews. They can be built ingame from 3 wooden sheets or placed by a mapper. They can also be deconstructed with a wrench into 3 wooden sheets. Pews are flammable.

This is my first PR, so please let me know if I messed up somewhere. Also, I did test and confirmed that it does compile.

## Why It's Good For The Game

Just adds some chapel themed chairs, basically. Its entirely an RP thing.

## Changelog
:cl:
add: Wooden pews may be constructed from 3 wood for all your religious needs!
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
why is own .dmi

**Comment by Caiggas** [comment]
I suppose I could merge it into something else. maybe chairs? I was following patterns established by similar items. Ratty couches have their own dmi, so I figured it was the correct way to do it. I apologize if I was incorrect. Let me know what dmi I should put this into, and Ill make the changes. Thanks for the feedback.

**Comment by ExcessiveUseOfCobblestone** [comment]
the same as couches should be fine, you can rename the file if it bothers you (just update code)

**Comment by Caiggas** [comment]
So, I apologize for my ignorance, but do I just close this PR, make my changes, and then open a new one? or is there a way for me to "overwrite" this PR with a corrected branch?

**Comment by ExcessiveUseOfCobblestone** [comment]
you just make the changes on your `Add-Chapel-Pews` branch and they'll get transferred to here.

**Comment by timothymtorres** [comment]
Those are some pretty nice sprites.  Well done!

**Comment by Caiggas** [comment]
Alright, so I merged pew.dmi into sofa.dmi and updated pew.dm. I then mapped a pew in and tested it to confirm everything still works.

**Comment by Caiggas** [comment]
so, travis ci fails on the "build all maps" step. I don't really understand what this is... Can anyone help me fix this? Or is this not something I need to worry about?

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -205,6 +205,13 @@ GLOBAL_LIST_INIT(wood_recipes, list ( \
 	new/datum/stack_recipe("wooden crate", /obj/structure/closet/crate/wooden, 6, time = 50, one_per_turf = TRUE, on_floor = TRUE),\
 	new/datum/stack_recipe("baseball bat", /obj/item/melee/baseball_bat, 5, time = 15),\
 	new/datum/stack_recipe("loom", /obj/structure/loom, 10, time = 15, one_per_turf = TRUE, on_floor = TRUE), \
+	null, \
```
Comment (optimumtact) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: what's with the null entries?

File: code/game/objects/structures/beds_chairs/pew.dm [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+/obj/structure/chair/pew/left
+	name = "left wooden pew end"
+	icon_state = "pewend_left"
+	var/mutable_appearance/leftpewarmrest
+
+/obj/structure/chair/pew/left/Initialize()
+	leftpewarmrest = GetLeftPewArmrest()
+	leftpewarmrest.layer = ABOVE_MOB_LAYER
+	return ..()
```
Comment (optimumtact) [inline code/game/objects/structures/beds_chairs/pew.dm]: you can just do `. = ..()`

`.` is the default return value of a proc

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 1 above)
Comment (Caiggas) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: They add lines in the build menu to group items in the build menu. I copied the syntax from the sofa build options higher up in the file. I can remove them if you want.

Comment (optimumtact) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: no that's fine, didn't realise they were used for formatting

File: code/game/objects/structures/beds_chairs/pew.dm (hunk 2 above)
Comment (Caiggas) [reply code/game/objects/structures/beds_chairs/pew.dm]: gotcha, thanks for the info. I copied that syntax from comfy chairs.

Comment (Caiggas) [reply code/game/objects/structures/beds_chairs/pew.dm]: to be honest, I feel like there has to be a way to write that code in the pew section, and have it work in pew/left and pew/right, but I don't fully understand byond syntax all that well. I feel like having essentially identical code in pew/left and pew/right is dirty, but I wan't sure how to do it better.

**Comment by PKPenguin321** [comment]
Nice. Always found it weird that the chapel had plain metal chairs all over the place

**Comment by Caiggas** [comment]
same here, I was mapping for fun, and I realized that the chapel didn't feel like a religious space.

File: code/game/objects/structures/beds_chairs/pew.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/structures/beds_chairs/pew.dm]:
```DM
//under the base type
var/armrest_mut_appearance_key
var/mutable_appearance/the_armrest

//under the subtypes
armrest_mut_appearance_key = "pewend_left_armrest" //or "pewend_right_armrest" 

//Procs
/obj/structure/chair/pew/Initialize()
    . = ..()
    if(armrest_mut_appearance_key)
        the_armrest = mutable_appearance('icons/obj/sofa.dmi', armrest_mut_appearance_key)
        the_armrest.layer = ABOVE_MOB_LAYER

/obj/structure/chair/pew/Destroy()
    if(the_armrest)
        QDEL_NULL(the_armrest)
    return ..()

/obj/structure/chair/pew/post_buckle_mob(mob/living/M)
	. = ..()
	update_armrest()

/obj/structure/chair/pew/post_unbuckle_mob()
	. = ..()
	update_armrest()

/obj/structure/chair/pew/proc/update_armrest()
    if(the_armrest)
        if(has_buckled_mobs())
		    add_overlay(the_armrest)
	    else
		    cut_overlay(the_armrest)
```

**Comment by Jimmius** [comment]
can't wait for the upcoming chapel rework where it has it's own minigames and systems that interact with the rest of the station, rather than just being a cool place to cremate bodies

**Comment by JJRcop** [comment]
In this case travis failed on cdd4cc165dc6e6d038ec4e0f3ee555ec6488e52e because of an internal error unrelated to your code, however in the future, you can click on each build job and get a log from the worker, so you can see the compile errors that had caused it to fail. This is a very valuable resource to remember. You can check appveyor too. If it fails on travis it probably won't pass on appveyor either.

**Comment by Cyprex** [comment]
The part between the CL is a summary of your changes for players, as such, consider something like "You can now build wooden benches for all your religious needs with 3 wood".

**Comment by optimumtact** [comment]
The changelog should be a summary for the players and try to avoid putting any weird formatting in it. It doesn't handle much outside of plain text.

**Comment by Caiggas** [comment]
Fixed it, thanks

**Comment by JJRcop** [comment]
The add labels must be in the cl tags, you shouldn't put anything else in there besides tags. Here's a legend of the available tags: https://github.com/tgstation/tgstation/blob/master/.github/PULL_REQUEST_TEMPLATE.md#changelog

**Comment by ExcessiveUseOfCobblestone** [comment]
someone edit this to have my code it's as close to free GBP as you're gonna get lol

**Comment by Caiggas** [comment]
Yeah, I'm gonna update it when I get the motivation. What is GBP? My brain is drawing a blank.

**Comment by ExcessiveUseOfCobblestone** [comment]
it's coder Good Boy Points

**Pull request merged**